### ISSUE TITLE
Assertion Faliure

### ISSUE THREAD

**Issue by harshvishu** [fuzz-productions/FZAccordionTableView#22]
Crashing after updating to latest version 0.2.2

```
Assertion failure in -[FZAccordionTableView tappedHeaderView:], /Users/.........................................................Pods/FZAccordionTableView/FZAccordionTableView/FZAccordionTableView.m:259
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: sectionHeaderView'
```

**Can this be because of the following line?**

`tableView.tableFooterView = UIView()`

**Comment by kgaidis** [comment]
When you leave `tableFooterView` to `nil` does it not crash anymore?

Could you please give steps to reproduce the issue? Including iOS version, device etc.

The bug could be anything. Without steps to reproduce or a point to why the issue is happening it's not easy to debug.

**Comment by harshvishu** [comment]
I am not sure how to reproduce the bug as it is inconsistent . This seems to be fixed after I updated the code to:

`tableView.tableFooterView = FZAccordionTableViewHeaderView()
`

As for your question '_When you leave tableFooterView to nil does it not crash anymore?_'.
- Yes, when I don't use any footer view it does not crashes.

FZAccordionTableView Version: 0.2.2
Xcode Version: 7.3.1 (7D1014)
iOS/OSX version: OS X El Captain 10.11.4 (15E65)
Dependency manager Version: cocoapods v1

**Comment by harshvishu** [comment]
Update

`tableView.tableFooterView = FZAccordionTableViewHeaderView()`

This doesn't fix the issue. Issue appears to be random. So I can't tell you how to reproduce this.

**Comment by kgaidis** [comment]
Hi, thank you for the updates.

I tried to use the Swift example you can find in this repository and set the `tableFooterView` like this:

```
let footerView = UIView()
footerView.frame = CGRectMake(0, 0, self.view.bounds.width, 40.0)
footerView.backgroundColor = UIColor.redColor()
tableView.tableFooterView = footerView
```

However, didn't find any crashes after tapping everywhere. 

I assume this happens when you are tapping, and not when calling `toggleSection:`?

There are two cases where `tappedHeaderView:` can be called:
1. User tapped the header view.
2. A call is made to `toggleSection`.

Could you provide the whole stack trace?

Also, it would be helpful if you could try to replicate this scenario using the Swift example in the repository. Using a simple example can help to isolate the problem. As of right now, setting the `tableFooterView` in the example doesn't cause any crashes that I see.

**Comment by kgaidis** [comment]
My initial suspicion would be that `toggleSection:` is being called, and a wrong index is being passed to it. But hard to tell with no code or a stack trace.

**Comment by harshvishu** [comment]
I'll try to reproduce the exception & post the stack trace.

**Comment by hanley91** [comment]
I have some insight into this issue... 
- (void)tappedHeaderView:(FZAccordionTableViewHeaderView *)sectionHeaderView {
  -----> NSParameterAssert(sectionHeaderView);

For some reason this assert is causing issues. Commenting it out seems to fix it, so i think the assertion is not receiving what it is expecting, or something of that nature.

**Comment by kgaidis** [comment]
Hi there, the assert statement is there for a reason (the method should not get called without a non-nil parameter), commenting it out would just hide a bug in the code, so I'd be interested in seeing the stack trace of what caused the crash if there is one available, thank you

**Comment by kgaidis** [comment]
For future refrence, seems like the problem people are experiencing is that when calling `toggleSection:` the section is actually not visible, which means that the headerview does not exist, which is why the assertion is firing. Full credits to @ftp27 with his pull request:
https://github.com/fuzz-productions/FZAccordionTableView/pull/24

For now, you can get around this problem by wrapping `toggleSection:` calls like this:

```
CGRect rect = [self.tableView rectForHeaderInSection:section];
[self.tableView scrollRectToVisible:rect animated:NO];
dispatch_async(dispatch_get_main_queue(), ^{
   [self.tableView toggleSection:section];
});
```

I am working on changing the code to make `toggleSection:` work without the need to scroll.

**Comment by kgaidis** [comment]
Fixed in `0.2.3`.
